torch/csrc/jit/codegen/cuda/scheduler/transpose.cpp

```diff
@@ -432,10 +556,65 @@ void scheduleTranspose(Fusion* fusion, const TransposeParams& params) {
   auto inner_most_id1 = scheduler_utils::innerMostRootDim(reference1);
   auto inner_most_id2 = scheduler_utils::innerMostRootDim(reference2);
 
-  auto inner_most_pos1_in_ref1 =
+  //////////////////////////////////////////
+  // Step 1: Make virtual inner most dims //
+  //////////////////////////////////////////
+
+  // See note [Supporting small transpose dimensions]
+
+  // split big dims so that we have enough dimensions available to merge with
+  // inner-most dims to create the virtual inner-most dim
+  scheduler_utils::splitDims(reference1, params.split_before_tiling);
+  auto merged1 = scheduler_utils::mergeDims(
+      reference1, params.dims_merged_with_1, params.dims_merged_with_2);
+  std::vector<size_t> merged1_vec;
+  if (merged1.has_value()) {
+    merged1_vec.push_back(*merged1);
+  }
+  auto merged2 = scheduler_utils::mergeDims(
```

Comment:
Nit comment: Changing the updated value to where that final merged dim is from the first merge step, so know which dims are the results of the merge. Merge step returns the resulting merged dims.